test_tf2/test/test_message_filter.cpp

```diff
@@ -114,109 +127,152 @@ geometry_msgs::TransformStamped createTransform(Quaternion q, Vector3 v, builtin
+  filter.registerCallback(std::bind(&Notification::notify, &n, std::placeholders::_1));
+
+  builtin_interfaces::msg::Time stamp = tf2_ros::toMsg(tf2::timeFromSec(1));
+  buffer.setTransform(createTransform(tf2::Quaternion(0,0,0,1), tf2::Vector3(1,2,3), stamp, "frame1", "frame2"), "me");
 
-  geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
+  std::shared_ptr<geometry_msgs::msg::PointStamped> msg = std::make_shared<geometry_msgs::msg::PointStamped>();
   msg->header.stamp = stamp;
   msg->header.frame_id = "frame2";
 
   filter.add(msg);
 
   EXPECT_EQ(1, n.count_);
+
+  node.reset();
 }
 
 TEST(MessageFilter, postTransforms)
 {
-  BufferCore bc;
+  auto node = rclcpp::Node::make_shared("tf2_ros_message_filter");
+  auto create_timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    node->get_node_base_interface(),
+    node->get_node_timers_interface());
+
+  rclcpp::Clock::SharedPtr clock = std::make_shared<rclcpp::Clock>(RCL_SYSTEM_TIME);
+  tf2_ros::Buffer buffer(clock);
+  buffer.setCreateTimerInterface(create_timer_interface);
+  tf2_ros::MessageFilter<geometry_msgs::msg::PointStamped> filter(buffer, "frame1", 10, node);
   Notification n(1);
-  MessageFilter<geometry_msgs::PointStamped> filter(bc, "frame1", 1, 0);
-  filter.registerCallback(boost::bind(&Notification::notify, &n, _1));
 
-  builtin_interfaces::msg::Time stamp(1);
+  filter.registerCallback(std::bind(&Notification::notify, &n, std::placeholders::_1));
+
+  builtin_interfaces::msg::Time stamp = tf2_ros::toMsg(tf2::timeFromSec(1));
 
-  geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
+  std::shared_ptr<geometry_msgs::msg::PointStamped> msg = std::make_shared<geometry_msgs::msg::PointStamped>();
   msg->header.stamp = stamp;
   msg->header.frame_id = "frame2";
 
   filter.add(msg);
 
   EXPECT_EQ(0, n.count_);
 
-  bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp, "frame1", "frame2"), "me");
+  buffer.setTransform(createTransform(tf2::Quaternion(0,0,0,1), tf2::Vector3(1,2,3), stamp, "frame1", "frame2"), "me");
 
   EXPECT_EQ(1, n.count_);
+  node.reset();
 }
-
-TEST(MessageFilter, queueSize)
-{
-  BufferCore bc;
-  Notification n(10);
-  MessageFilter<geometry_msgs::PointStamped> filter(bc, "frame1", 10, 0);
-  filter.registerCallback(boost::bind(&Notification::notify, &n, _1));
-  filter.registerFailureCallback(boost::bind(&Notification::failure, &n, _1, _2));
-
-  builtin_interfaces::msg::Time stamp(1);
-
-  for (int i = 0; i < 20; ++i)
-  {
-    geometry_msgs::PointStampedPtr msg(new geometry_msgs::PointStamped);
-    msg->header.stamp = stamp;
-    msg->header.frame_id = "frame2";
-
-    filter.add(msg);
-  }
-
-  EXPECT_EQ(0, n.count_);
-  EXPECT_EQ(10, n.failure_count_);
-
-  bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp, "frame1", "frame2"), "me");
-
-  EXPECT_EQ(10, n.count_);
-}
+// TODO (ahcorde): enable this when this is available
```

Comment:
Can you be a bit more descriptive on what is missing to be able to port this test?